### PR TITLE
added netpol section

### PR DESCRIPTION
**Pull request by vivekratan88**
added netpolicy and explanation

### CODE REVIEW

File: src/docs/app-monitoring/user-defined-monitoring.md [hunk 1]
```diff
@@ -194,6 +194,59 @@ oc apply -f app-alerting-rule.yaml
 
 `AlertmanagerConfig` is part of the Prometheus Operator, which helps manage Prometheus instances and configurations in Kubernetes. Using AlertmanagerConfig ,allows you to manage and version control your alerting rules and configurations within Kubernetes resources, keeping them aligned with your application code and infrastructure. More information about [Alertmanager](https://developer.gov.bc.ca/docs/default/component/platform-developer-docs/docs/platform-automation/alertmanager/)
 
+Please have this network policy added to ensure that proper metrics are scraped
+
+```yaml
+- apiVersion: v1
+    kind: Service
```
Comment (StevenBarre) [inline src/docs/app-monitoring/user-defined-monitoring.md:201]: There's already an example Service under `Expose the metrics from your app` section.

File: src/docs/app-monitoring/user-defined-monitoring.md [hunk 2]
```diff
@@ -194,6 +194,59 @@ oc apply -f app-alerting-rule.yaml
...
+      selector:
+        app: prometheus-metrics-app
+      ports:
+        - name: metrics
+          protocol: TCP
+          port: 8000
+          targetPort: 8000
+
+  - apiVersion: monitoring.coreos.com/v1
+    kind: ServiceMonitor
```
Comment (StevenBarre) [inline src/docs/app-monitoring/user-defined-monitoring.md:217]: There's already an example ServiceMonitor under the `Create a ServiceMonitor` section of the doc

File: src/docs/app-monitoring/user-defined-monitoring.md [hunk 3]
```diff
@@ -194,6 +194,59 @@ oc apply -f app-alerting-rule.yaml
...
+    spec:
+      endpoints:
+        - interval: 30s
+          port: metrics
+          scheme: http
+      selector:
+        matchLabels:
+          app: prometheus-metrics
+
+  - apiVersion: networking.k8s.io/v1
```
Comment (StevenBarre) [inline src/docs/app-monitoring/user-defined-monitoring.md:232]: Since this isn't part of a kind: List anymore, we don't need to indent it

**Review by StevenBarre — CHANGES_REQUESTED** [review]

File: src/docs/app-monitoring/user-defined-monitoring.md (hunk 3 above)
Comment (vivekratan88) [reply src/docs/app-monitoring/user-defined-monitoring.md:232]: removed - and aligned the podselector and policytypes as part of spec mapping not the ingress list

File: src/docs/app-monitoring/user-defined-monitoring.md (hunk 2 above)
Comment (vivekratan88) [reply src/docs/app-monitoring/user-defined-monitoring.md:217]: removed this redudant section

File: src/docs/app-monitoring/user-defined-monitoring.md (hunk 1 above)
Comment (vivekratan88) [reply src/docs/app-monitoring/user-defined-monitoring.md:201]: removed this redundant section

File: src/docs/app-monitoring/user-defined-monitoring.md [hunk 4]
```diff
@@ -194,6 +194,27 @@ oc apply -f app-alerting-rule.yaml
...
 `AlertmanagerConfig` is part of the Prometheus Operator, which helps manage Prometheus instances and configurations in Kubernetes. Using AlertmanagerConfig ,allows you to manage and version control your alerting rules and configurations within Kubernetes resources, keeping them aligned with your application code and infrastructure. More information about [Alertmanager](https://developer.gov.bc.ca/docs/default/component/platform-developer-docs/docs/platform-automation/alertmanager/)
 
+Please have this network policy added to ensure that proper metrics are scraped
+
+```yaml
+  apiVersion: networking.k8s.io/v1
+  kind: NetworkPolicy
+  metadata:
+    name: allow-from-openshift-monitoring
+    namespace: be1c6b-test
```
Comment (StevenBarre) [inline src/docs/app-monitoring/user-defined-monitoring.md:204]:
```suggestion
    namespace: license-dev
```
Match the namespace used in the rest of the document

File: src/docs/app-monitoring/user-defined-monitoring.md [hunk 5]
```diff
@@ -194,6 +194,27 @@ oc apply -f app-alerting-rule.yaml
...
+      - from:
+          - namespaceSelector:
+              matchLabels:
+                kubernetes.io/metadata.name: openshift-user-workload-monitoring
+    podSelector: {}
+    policyTypes:
+      - Ingress
+```
+
+Here we can see that label under `service` should match with the `ServiceMonitor` under selector and matchLabels.
```
Comment (StevenBarre) [inline src/docs/app-monitoring/user-defined-monitoring.md:216]: I don't think we need this sentence now that the service and servicemonitor yaml was removed? Or maybe it should be higher up in the doc where those objects are described.

**Review by StevenBarre — CHANGES_REQUESTED** [review]

File: src/docs/app-monitoring/user-defined-monitoring.md (hunk 5 above)
Comment (vivekratan88) [reply src/docs/app-monitoring/user-defined-monitoring.md]: removed

File: src/docs/app-monitoring/user-defined-monitoring.md (hunk 4 above)
Comment (vivekratan88) [reply src/docs/app-monitoring/user-defined-monitoring.md]: modified as per feedback

**Review by StevenBarre — APPROVED** [review]

**Pull request merged**